pkg/consuldp/metrics_test.go

```diff
@@ -13,17 +13,43 @@ import (
 	"testing"
 	"time"
 
+	"github.com/armon/go-metrics"
 	"github.com/hashicorp/consul-dataplane/internal/bootstrap"
 	metricscache "github.com/hashicorp/consul-dataplane/pkg/metrics-cache"
 	"github.com/stretchr/testify/require"
 )
 
 var (
+	dogStatsdAddr    = "127.0.0.1"
 	envoyMetricsPort = 19000
 	envoyMetricsAddr = "127.0.0.1"
 	envoyMetricsUrl  = fmt.Sprintf("http://%s:%v/stats/prometheus", envoyMetricsAddr, envoyMetricsPort)
+
+	emptyTags = []metrics.Label{}
 )
 
+func setupTestServerAndBuffer(t *testing.T) (*net.UDPConn, []byte) {
+	udpAddr, err := net.ResolveUDPAddr("udp", fmt.Sprintf("%v:%v", dogStatsdAddr, 0))
+	if err != nil {
+		t.Fatal(err)
+	}
+	server, err := net.ListenUDP("udp", udpAddr)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return server, make([]byte, 1024)
+}
+
+func assertServerMatchesExpected(t *testing.T, server *net.UDPConn, buf []byte, expected string) {
+	t.Helper()
+	n, _ := server.Read(buf)
+	msg := string(buf[:n])
+
+	if msg != expected {
+		t.Fatalf("Line %s does not match expected: %s", msg, expected)
+	}
```

Comment:
nit: could this use `require.Equal`?

```suggestion
	require.Equal(t, expected, msg, "Line %s does not match expected: %s", msg, expected)
```